source/nodejs/adaptivecards/src/utils.ts

```diff
@@ -4,6 +4,25 @@ import * as Enums from "./enums";
 import * as Shared from "./shared";
 import { HostConfig } from "./host-config";
 
+export function formatString(s: string, ...args: any[]): string {
+    let result = s;
+
+    if (result) {
+        let regEx = /{(\d+)}/g;
+        let matches: RegExpExecArray | null;
+
+        while ((matches = regEx.exec(result)) != null) {
```

Comment:
I know that this is mainly just for our own internal use, but it would be nice if this method were a *bit* more robust -- in particular:

```javascript
const templateString = "This is my {0} string.";
foo.formatString(templateString, "format", "extra"); // should probably throw an exception
```

```javascript
const templateString = "My {0} string.";
foo.formatString(templateString, "{1}", 8); // do we want this sort of chaining to work?
```

```javascript
const templateString = "{0} is my {1} {2}.";
foo.formatString(templateString, "This", "template"); // should probably throw an exception
```

```javascript
const templateString = "My {0} string.";
foo.formatString(templateString, { foo: "bar" }); // maybe an exception here? I'm not sure...

Reply:
This was designed to be resilient and explicitly not throw in those cases. But as per Risheek's suggestion, I'm moving to resource functions, so this is moot.